hibernate-core/src/main/java/org/hibernate/tool/schema/internal/ColumnDefinitions.java

```diff
@@ -26,7 +26,11 @@
 
 class ColumnDefinitions {
 
-	static boolean hasMatchingType(Column column, ColumnInformation columnInformation, Metadata metadata, Dialect dialect) {
+	static boolean hasMatchingType(Column column, ColumnInformation columnInformation, Metadata metadata, Dialect dialect, boolean forUpdate) {
+		if ( forUpdate && column.getColumnDefinition() != null ) {
+			// validation shouldn't use columnDefinition
```

Comment:
Why should `SchemaValidator` should ignore the `columnDefinition` if it's so important that `SchemaUpdate` take it into account?

Reply:
The original reason is it break some existing dialect specific tests, caused by JDBC driver doesn't return the type defined in `columnDefinition`, It's hard or impossible to not break existing behavior.
Take a step back, I think it's reasonable to make it lenient for validation schema, the purpose of `jakarta.persistence.schema-generation.database.action=validation` is to maintain schema manually but validate its compatibility with entity mappings, it shouldn't take `columnDefinition` into account.

Reply:
I mean, you can in principle change the tests if that's necessary.

But if it makes sense to consider `columnDefinition` for migration, it also makes sense to consider it for validation.

Reply:
> But if it makes sense to consider `columnDefinition` for migration, it also makes sense to consider it for validation.

I will give it a try if you insist on that.